objectstore-server/src/observability.rs

```diff
@@ -33,7 +33,19 @@ pub fn maybe_initialize_sentry(config: &Config) -> Option<sentry::ClientInitGuar
         environment: config.environment.clone(),
         server_name: config.server_name.clone(),
         sample_rate: config.sample_rate,
-        traces_sample_rate: config.traces_sample_rate,
+        traces_sampler: {
+            let propagate_traces = config.propagate_traces;
+            let traces_sample_rate = config.traces_sample_rate;
+            Some(std::sync::Arc::new(move |ctx| {
+                if propagate_traces && ctx.operation() == "http.server" {
+                    1.0
```

Comment:
Are you sure we want this behavior? In a high-throughput context this could be surprising - you might still need to set a sample rate even for the server "segments". 

I think need behavior closer to this:

```rs
if let Some(sampled) = ctx.sampled() && inherit_sampling_decision {
  f32::from(sampled)
} else {
  traces_sample_rate
}
```

Note that this will break extrapolation, since the SDK will report a sampling rate of `1.0` to the server. That will need to be fixed in the SDK.